src/odemis/gui/model/tab_gui_data.py

```diff
@@ -268,13 +290,17 @@ def select_current_position_feature(self):
         Given current stage position, either select one of the features closest to
           the position or create a new one with the position.
         """
-        current_position = self.main.stage.position.value
+        current_position = self.main.stage_bare.position.value
         current_feature = self.main.currentFeature.value
 
         def dist_to_pos(feature):
-            return math.hypot(feature.pos.value[0] - current_position["x"],
-                              feature.pos.value[1] - current_position["y"])
-
+            pm = self.main.posture_manager
+            position = get_feature_position_at_posture(pm, feature, pm.current_posture.value)
+            pos = pm.to_sample_stage_from_stage_position(position)
+            sample_stages_pos = pm.to_sample_stage_from_stage_position(current_position)
+            # Note: we can get the sample stage position direction from: self.stage.position.value
+            return math.hypot(pos["x"] - sample_stages_pos["x"],
```

Comment:
I think z should also be taken into account, because FM imaging can take place at different sample plane heigths

Reply:
in principle agree, but this is only used to filter when the user clicks a point in the GUI, which they can only do in XY. So there inst really a way to select a different "Z"